src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicRemoteExecutorTests.cs

```diff
@@ -20,27 +20,43 @@ public class MsQuicRemoteExecutorTests : QuicTestBase
         public MsQuicRemoteExecutorTests()
             : base(null!) { }
 
-        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
-        public void SslKeyLogFile_IsCreatedAndFilled()
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void SslKeyLogFile_IsCreatedAndFilled(bool enabledBySwitch)
         {
-            if (PlatformDetection.IsReleaseLibrary(typeof(QuicConnection).Assembly))
+            if (PlatformDetection.IsDebugLibrary(typeof(QuicConnection).Assembly) && !enabledBySwitch)
             {
-                throw new SkipTestException("Retrieving SSL secrets is not supported in Release mode.");
+                throw new SkipTestException("AppCtxSwitch is not necessary for SSLKEYLOGFILE in Debug.");
             }
 
             var psi = new ProcessStartInfo();
             var tempFile = Path.GetTempFileName();
             psi.Environment.Add("SSLKEYLOGFILE", tempFile);
 
-            RemoteExecutor.Invoke(async () =>
+            RemoteExecutor.Invoke(async (enabledBySwitch) =>
             {
+                if (bool.Parse(enabledBySwitch))
+                {
+                    AppContext.SetSwitch("System.Net.Security.EnableSslKeyLogging", true);
+                }
+
                 (QuicConnection clientConnection, QuicConnection serverConnection) = await CreateConnectedQuicConnection();
                 await clientConnection.DisposeAsync();
                 await serverConnection.DisposeAsync();
-            }, new RemoteInvokeOptions { StartInfo = psi }).Dispose();
+            }
+            , enabledBySwitch.ToString(), new RemoteInvokeOptions { StartInfo = psi }).Dispose();
 
-            Assert.True(File.Exists(tempFile));
-            Assert.True(File.ReadAllText(tempFile).Length > 0);
+            if (enabledBySwitch)
+            {
+                Assert.True(File.Exists(tempFile));
+                var text = File.ReadAllText(tempFile);
```

Comment:
Any reason for the extra variable here?
I'm wondering why the content of the if-else branches differs so much if it tests the exact opposite.